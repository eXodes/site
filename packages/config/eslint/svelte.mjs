import { rules } from "./rules.mjs";
import eslintConfigPrettier from "eslint-config-prettier";
import eslintConfigTurbo from "eslint-config-turbo/flat";
import onlyWarn from "eslint-plugin-only-warn";
import eslintPluginSvelte from "eslint-plugin-svelte";
import eslintPluginUnicorn from "eslint-plugin-unicorn";
import globals from "globals";
import svelteParser from "svelte-eslint-parser";
import tsEslint from "typescript-eslint";

export default tsEslint.config(
  eslintPluginUnicorn.configs["flat/recommended"],
  ...tsEslint.configs.recommended,
  ...tsEslint.configs.stylistic,
  ...eslintPluginSvelte.configs["flat/recommended"],
  eslintConfigPrettier,
  ...eslintPluginSvelte.configs["flat/prettier"],
  ...eslintConfigTurbo,
  {
    plugins: {
      "only-warn": onlyWarn,
    },

    languageOptions: {
      globals: { ...globals.browser, ...globals.node },

      ecmaVersion: "latest",
      sourceType: "module",

      parser: tsEslint.parser,
      parserOptions: {
        tsconfigRootDir: import.meta.dirname,
        parser: tsEslint.parser,
      },
    },

    rules: {
      ...rules,
    },
  },
  {
    files: ["**/*.svelte"],

    languageOptions: {
      ecmaVersion: 5,
      sourceType: "script",

      parser: svelteParser,
      parserOptions: {
        parser: tsEslint.parser,
        extraFileExtensions: [".svelte"],
      },
    },
  }
);
