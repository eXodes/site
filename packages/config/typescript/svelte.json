{
  "display": "Svelte",
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "lib": ["esnext", "dom", "dom.iterable"],
    "module": "esnext",
    "moduleDetection": "force",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext"
  }
}
