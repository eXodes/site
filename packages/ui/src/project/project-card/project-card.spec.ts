import ProjectCard, { type Project } from "./project-card.svelte";
import { render } from "@testing-library/svelte";
import { describe, expect, test } from "vitest";

describe("ProjectCard", () => {
  test("should render component", async () => {
    const project: Project = {
      title: "Project 1",
      description: "A brief description of the project 1.",
      company: "Company 1",
      image: "https://via.placeholder.com/150",
      link: "https://example.com",
      year: 2021,
    };

    const { container } = render(ProjectCard, { project });

    expect(container.children).toMatchSnapshot();
  });
});
