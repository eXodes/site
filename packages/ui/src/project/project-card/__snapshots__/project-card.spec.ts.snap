// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectCard > should render component 1`] = `
HTMLCollection [
  <a
    href="https://example.com"
    target="_blank"
  >
    <img
      alt="Project 1"
      class="aspect-video w-full rounded-t-lg object-cover lg:rounded-lg"
      src="https://via.placeholder.com/150"
    />
     
    <div
      class="group rounded-lg bg-gradient-to-b from-transparent from-50% to-yellow/90 to-50% bg-[length:100%_200%] transition-all duration-500 hover:bg-[0_100%] hover:backdrop-blur lg:absolute lg:inset-0"
    >
      <div
        class="flex flex-col items-center justify-center gap-4 bg-yellow p-4 text-center transition-all duration-500 group-hover:translate-y-0 lg:absolute lg:inset-0 lg:translate-y-[100%] lg:bg-transparent"
      >
        <div>
          <h3
            class="text-xl font-semibold"
          >
            Project 1
          </h3>
           
          <p
            class="text-xs uppercase tracking-wider text-gray-600"
          >
            Company 1
          </p>
           
          <p
            class="px-2 py-1 text-sm tracking-wide text-gray-500"
          >
            2021
          </p>
        </div>
         
        <div
          class="lg:px-6"
        >
          <p
            class="text-xs text-gray-700"
          >
            A brief description of the project 1.
          </p>
        </div>
         
        <!---->
      </div>
    </div>
  </a>,
]
`;
